server/models/services/GithubWorkflowService.js

```diff
@@ -0,0 +1,157 @@
+const GITHUB_APP_INSTALLATION_ID = '42217598';
+
+// > The time that the JWT was created. To protect against clock drift, we
+// > recommend that you set this 60 seconds in the past and ensure that your
+// > server's date and time is set accurately (for example, by using the Network
+// > Time Protocol).
+//
+// https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app
+const calculateIssuedAt = () => Math.round(Date.now() / 1000) - ONE_MINUTE;
+
+const createJWT = (payload, privateKey, algorithm) => {
+    return new Promise((resolve, reject) => {
+        jwt.sign(payload, privateKey, { algorithm }, (err, token) => {
+            token ? resolve(token) : reject(err);
+        });
+    });
+};
+
+// > The expiration time of the JWT, after which it can't be used to request an
+// > installation token. The time must be no more than 10 minutes into the
+// > future.
+//
+// https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app
+const calculateExpiresAt = () => Math.round(Date.now() / 1000) + 9 * ONE_MINUTE;
+
+const fetchInstallationAccessToken = async (jsonWebToken, installationID) => {
+    const response = await fetch(
+        `https://api.github.com/app/installations/${installationID}/access_tokens`,
+        {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.github+json',
+                Authorization: `Bearer ${jsonWebToken}`,
+                'X-GitHub-Api-Version': '2022-11-28'
+            }
+        }
+    );
+
+    if (!response.ok) {
+        const message = await (() => {
+            try {
+                return response.text();
+            } catch (_) {
+                return 'Unable to retrieve installation access token';
+            }
+        })();
+
+        throw new Error(message);
+    }
+
+    return (await response.json()).token;
+};
+
+const createGithubWorkflow = async ({
+    job,
+    directory = 'alert',
+    gitSha = 'master'
+}) => {
+    const payload = {
+        iat: calculateIssuedAt(),
+        exp: calculateExpiresAt(),
+        iss: GITHUB_APP_ID
+    };
+    const jsonWebToken = await createJWT(payload, privateKey, ALGORITHM);
+    const accessToken = await fetchInstallationAccessToken(
+        jsonWebToken,
+        GITHUB_APP_INSTALLATION_ID
+    );
+    const inputs = {
+        nonce: job.id,
+        callback_url: `https://${process.env.AUTOMATION_CALLBACK_FQDN}/api/jobs/${job.id}/result`,
+        status_url: `https://${process.env.AUTOMATION_CALLBACK_FQDN}/api/jobs/${job.id}/update`,
+        // TODO: This should probably just be configured instead of us passing it to the scheduler every time
+        callback_header: `x-automation-secret:${process.env.AUTOMATION_SCHEDULER_SECRET}`,
```

Comment:
It seems to me that passing it every time alleviates us from the need to synchronize a setting between two systems (ARIA-AT App and GitHub Actions). Is that right? What are the drawbacks?

Reply:
I think my goal with putting this here instead of synced between the two servers was some day to have this `x-automation-secret` be a secret key per job instead of globally.  As it works now, (for MVP) we could define this key as a permanent secret in the github action repo and remove it from here entirely (assuming we use the same key across all environments we want to be able to actively test the github workflow against)

I doubt there are actors trying to sabotage our test results, but by exchanging the key, and having it be individual per job should greatly reduce the ability for someone to corrupt/fake/hack the data collection process.

Reply:
It sounds like we are in agreement, after all! Let's remove the "TODO" comment because we don't actually desire the configuration-based approach it suggests.

Reply:
TODO removed